libs/auth/react/src/lib/Authenticator/CheckIdpSession.tsx

```diff
@@ -30,58 +22,108 @@ export interface SessionInfoResponse {
   isExpired?: boolean
 }
 
-interface SessionInfoRequest {
-  type: SessionInfoMessageType.SessionInfoRequest
+interface UserSession {
+  /* Flag to indicate if we have received session info message from the IDS iframe. */
+  hasMessage: boolean
+
+  /* The expected time when the user session is ending. */
+  sessionEnd: Date | null
+
+  /* The handle of the setInterval function. If set this indicates the user session active state. */
+  intervalHandle: ReturnType<typeof setInterval> | null
+
+  /* Number of times we have tried to load the iframe to receive a new session info message. */
+  retryCount: number
 }
 
-const messageEventName = 'message'
+const MAX_RETRIES = 2
+const MESSAGE_EVENT_NAME = 'message'
```

Comment:
IMO this is unneccessary obfuscation when dealing with a built in browser event.